parlai/core/torch_agent.py

```diff
@@ -769,7 +769,14 @@ def _invsqrt_lr(step):
             # first make sure there are no null pointers
             states = {}
 
-        if states and states.get('lr_scheduler_type') != self.opt['lr_scheduler']:
+        if (
+            # there is already an old LR scheduler saved on disk
+            states and
+            # and the old LR scheduler is different
+            states.get('lr_scheduler_type') != self.opt['lr_scheduler'] and
```

Comment:
only question... do we need an opt.get('lr_scheduler') here?

Reply:
A super old model might have been trained before we introduce lr_schedulers.

Reply:
then why didn't you put an opt.get there?

Reply:
Hmm, right above it will `raise ValueError` if it's not in the dictionary so i guess it's handled